pkg/controller/registry/grpc/source.go

```diff
@@ -100,10 +105,69 @@ func (s *SourceStore) Get(key registry.CatalogKey) *SourceConn {
 	return &source
 }
 
+func grpcProxyURL(addr string) (*url.URL, error) {
+	// Handle ip addresses
+	host, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := url.Parse(host)
+	if err != nil {
+		return nil, err
+	}
+
+	// Hardcode fields required for proxy resolution
+	url.Host = addr
+	url.Scheme = "http"
+
+	proxyConfig := httpproxy.FromEnvironment()
+
+	// Override HTTPS_PROXY and HTTP_PROXY with GRPC_PROXY
+	proxyConfig.HTTPProxy = getGRPCProxyEnv()
+	proxyConfig.HTTPSProxy = getGRPCProxyEnv()
+
+	// Check if a proxy should be used based on Environment variables
+	return proxyConfig.ProxyFunc()(url)
+}
+
+func getGRPCProxyEnv() string {
+	return getEnvAny("GRPC_PROXY", "grpc_proxy")
+}
+
+func getEnvAny(names ...string) string {
```

Comment:
Nit: Could this live in a utilities package somewhere? Seems like a useful function to be able to import in other places and reuse.

Reply:
Dave Cheney had a nice writeup on utility packages: https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common

To your point, this could be exported into a `environment` package, but I didn't see the value add since no other piece uses this yet.

Reply:
Ah, good point. I more meant making an `environment` package as opposed to a `util` or `common` package like you mentioned but I agree that the value add is low. Resolving.